### PR TITLE
Friend list (list + add + remove)

### PR DESCRIPTION
**Pull request by Tetardfurieux**

### CODE REVIEW

**Review by Aco-Hub — COMMENTED** [review]
Lack some comments. The code have to be better modularize. Some functions can be moved to the class database.

**Review by emonnin-epfl — CHANGES_REQUESTED** [review]
You can comment on your code to help others understand it better.

File: app/src/main/java/ch/epfl/sdp/healthplay/FriendList_Frag.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
+                fragmentTransaction.replace(R.id.fragmentContainerView, new Frag_Home());
+                fragmentTransaction.commit();
+            }
+        }
+        );
+
+        addFriendButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```
[the remark this answers is not in the document]
Comment (Tetardfurieux) [reply app/src/main/java/ch/epfl/sdp/healthplay/FriendList_Frag.java:106]: Done

File: app/src/main/res/layout/fragment_add_friend.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <Button
+            android:id="@+id/AddFriend"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="158dp"
+            android:layout_marginEnd="159dp"
+            android:layout_marginBottom="455dp"
+            android:text="Add friend"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/fragment_add_friend.xml:36]: "Add friend", should use `@string` resource

File: app/src/main/res/layout/fragment_add_friend.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/friendSearch"
+            app:layout_constraintVertical_bias="1.0" />
+
+        <Button
+            android:id="@+id/backButton"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:backgroundTint="?attr/gradientPrimary_color"
+            android:text="Go back"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/fragment_add_friend.xml:50]: "Go back", should use `@string` resource

File: app/src/main/res/layout/fragment_add_friend.xml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        android:layout_height="match_parent">
+
+        <EditText
+            android:id="@+id/friendSearch"
+            android:layout_width="411dp"
+            android:layout_height="191dp"
+            android:clickable="true"
+            android:cursorVisible="true"
+            android:focusable="true"
+            android:hint="Enter the name of your friend"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/fragment_add_friend.xml:20]: "Enter the name of your friend", should use `@string` resource

File: app/src/main/res/layout/activity_friend_list_item.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            android:text="friendName"
+            android:textAlignment="center"
+            android:textSize="24sp" />
+
+        <Button
+            android:id="@+id/removeFriendButton"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:backgroundTint="?attr/gradientPrimary_color"
+            android:text="Remove Friend" />
```
Comment (Aco-Hub) [inline app/src/main/res/layout/activity_friend_list_item.xml:40]: "Remove Friend", should use `@string` resource

File: app/src/main/res/layout/activity_friend_list_item.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        app:layout_constraintLeft_toRightOf="@+id/leftGuidelineItem"
+        app:layout_constraintRight_toLeftOf="@+id/rightGuidelineItem"
+        app:layout_constraintTop_toBottomOf="@+id/upGuidelineItem"
+        app:layout_constraintBottom_toTopOf="@+id/downGuidelineItem">
+
+        <ImageView
+            android:id="@+id/friendProfilePicture"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:contentDescription="Profile picture of your friend"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/activity_friend_list_item.xml:23]: "Profile picture of your friend", should use `@string` resource

File: app/src/main/res/layout/activity_friend_list_item.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            app:srcCompat="@drawable/rounded_logo" />
+
+        <TextView
+            android:id="@+id/friendName"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:contentDescription="Name of your friend"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/activity_friend_list_item.xml:30]: "Name of your friend", should use `@string` resource

File: app/src/main/res/layout/activity_friend_list_item.xml [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+            android:text="friendName"
```
Comment (Aco-Hub) [inline app/src/main/res/layout/activity_friend_list_item.xml:31]: "friendName", should use `@string` resource

File: app/src/main/java/ch/epfl/sdp/healthplay/FriendList_Frag.java [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+        View view = inflater.inflate(R.layout.fragment_friend_list_, container, false);
+        Button calendarButton = view.findViewById(R.id.friendToCalendar);
+        Button addFriendButton = view.findViewById(R.id.addFriendBouton);
+        ListView listView = view.findViewById(R.id.friendList);
+
+
+        calendarButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
```
[the remark this answers is not in the document]
Comment (Tetardfurieux) [reply app/src/main/java/ch/epfl/sdp/healthplay/FriendList_Frag.java:97]: Function created !

**Review by Aco-Hub — APPROVED** [review]

**Review by emonnin-epfl — APPROVED** [review]

**Pull request merged**